kong/plugins/basic-auth/access.lua

```diff
@@ -39,9 +38,17 @@ local function retrieve_credentials(request, header_name, conf)
     if m and m[1] then
       local decoded_basic = ngx.decode_base64(m[1])
       if decoded_basic then
-        local basic_parts = utils.split(decoded_basic, ":")
-        username = basic_parts[1]
-        password = basic_parts[2]
+        local basic_parts, err = ngx.re.match(decoded_basic, "([^:]+):(.+)")
```

Comment:
Nice, thanks! A few brief thoughts: we may want to localize the "ngx.re.match" function, and we will want to use the "oj" flags as the third param in this call to leverage PCRE JITing and cache the compiled expression. Thanks again!

Reply:
Good call, I'll apply those changes too